### PR TITLE
Session and filter are implemented

### PR DESCRIPTION
**Pull request by andromash**

### CODE REVIEW

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDB = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect name or password"));
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]:
```suggestion
                new AuthenticationException("Incorrect login or password"));
```

File: src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        String url = req.getServletPath();
+        if (url.equals("/login") || url.equals("/registration")) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null || userService.get(userId) == null) {
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/web/filter/AuthenticationFilter.java]:
```suggestion
        if (userId == null) {
```

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

**Review by dariiabutorina — APPROVED** [review]

**Review by dariiabutorina — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authService.login(login, pwd);
+            HttpSession httpSession = req.getSession();
+            httpSession.setAttribute("user_id", user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/LoginController.java]: let's make `user_id` constant

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -40,4 +40,11 @@ public User update(User user) {
     public boolean delete(Long id) {
         return Storage.users.removeIf(user -> user.getId().equals(id));
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream()
+                .filter(s -> s.getLogin().equals(login))
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]: `s` is not informative

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**